eo-maven-plugin/src/main/java/org/eolang/maven/footprint/CacheVersion.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2023 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.footprint;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Cache version.
+ *
+ * @since 0.30
+ */
+final class CacheVersion {
+
+    /**
+     * Not cacheable versions.
+     */
+    private static final String[] NOT_CACHEABLE = {"0.0.0", "SNAPSHOT"};
+
+    /**
+     * Version of the eo-maven-plugin.
+     */
+    private final String version;
+
+    /**
+     * Hash of the object.
+     */
+    private final String hash;
+
+    /**
+     * Ctor.
+     * @param ver Version of the eo-maven-plugin
+     * @param hsh Hash of the object
+     */
+    CacheVersion(final String ver, final String hsh) {
+        this.version = ver;
+        this.hash = hsh;
+    }
+
+    /**
+     * Checks if the version is cacheable.
+     * @return True if cacheable, false otherwise.
+     */
+    boolean cacheable() {
+        return Objects.nonNull(this.hash)
+            && !this.hash.isEmpty()
+            && (Objects.isNull(this.version)
+            || Arrays.stream(CacheVersion.NOT_CACHEABLE).noneMatch(this.version::contains));
+    }
+
+    /**
+     * Get cache path.
+     * @return Path.
+     */
+    Path cache() {
```

Comment:
@volodya-lombrozo I believe better name would be `path()`. `version.path()` seems to be more natural.